pkgs/applications/misc/phoc/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, fetchFromGitLab
+, meson
+, ninja
+, pkgconfig
+, wrapGAppsHook
+, libinput
+, gnome3
+, glib
+, gtk3
+, wayland
+, dbus
+, cmake
+, libdrm
+, libxkbcommon
+, wlroots
+}:
+
+stdenv.mkDerivation rec {
+  pname = "phoc";
+  version = "0.1.7";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "3d68b091d0075bd549f6fbb84d6c152f01a2354e";
+    sha256 = "1007kxap87h1bi79sz45lfb478d93pq0676cr5rnhbanljm1n28n";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    meson
+    ninja
+    pkgconfig
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    libdrm.dev
+    libxkbcommon
+    libinput
+    glib
+    gtk3
+    gnome3.gnome-desktop
+    gnome3.mutter
+    wayland
+    wlroots
+  ];
+
+  NIX_CFLAGS_COMPILE = "-I${libdrm.dev}/include/libdrm/";
```

Comment:
Please don't use `NIX_CFLAGS_COMPILE` - this should be passed to meson.

Reply:
I don't know what do you mean. Can you give me an example?

Reply:
Not sure if this is an upstream bug or nixpkgs one, the graphics stack is weird.

Reply:
I'm not sure what to do on this one

Reply:
If you could post the error message, it would be helpful.

Reply:
```
[...]
FAILED: src/25a6634@@phoc@sta/cursor.c.o 
gcc -Isrc/25a6634@@phoc@sta -Isrc -I../src -I/nix/store/vmd834p7b56ba3vmq53rsn5x1yqx77pi-libinput-1.15.5-dev/include -I/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/include -I/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/include/glib-2.0 -I/nix/store/9yfbxys7srn3cgx9jsmfcyakwpkwr9fs-glib-2.64.1/lib/glib-2.0/include -I/nix/store/q6ar7s3ld5l537pkjymwyz92a1cgv1qg-gtk+3-3.24.20-dev/include/gtk-3.0 -I/nix/store/18qmdsyy9f6xxxcxbk8zi7k7gl04z3r7-atk-2.36.0-dev/include/atk-1.0 -I/nix/store/n2amx937mh6k39iadvh567kq4g2ljly7-cairo-1.16.0-dev/include/cairo -I/nix/store/chrni3066nd9li7ccffs9yfqdhlqjdw7-freetype-2.10.1-dev/include/freetype2 -I/nix/store/chrni3066nd9li7ccffs9yfqdhlqjdw7-freetype-2.10.1-dev/include -I/nix/store/pj515z2qsfhwsp3izx0paxfj434hwjqn-gdk-pixbuf-2.40.0-dev/include/gdk-pixbuf-2.0 -I/nix/store/vhc7j53yqsvrr35whr178bh7spxpan6n-gsettings-desktop-schemas-3.36.1/include/gsettings-desktop-schemas -I/nix/store/hnigkkd0xdyl0byb09xxmx55r502acgs-pango-1.44.7-dev/include/pango-1.0 -I/nix/store/kicxnsx8wx5vdjdm1hxq13kkpykkb067-harfbuzz-2.6.4-dev/include/harfbuzz -I/nix/store/jsmq4mkqcklgpwdi34vp3b3g4mfy97z6-gnome-desktop-3.36.2-dev/include/gnome-desktop-3.0 -I/nix/store/5n9qf33f4ynngqjl4fi61ks9ccqhgbgi-pixman-0.38.4/include/pixman-1 -I/nix/store/fh1v53h0y0jcm2rd3cxkd5a8jf9hwp6p-wayland-1.18.0/include -I/nix/store/zvv99h24gav9ydnv1xh9gdlmxdscgkny-wlroots-0.10.1/include -I/nix/store/bwg82bn5m8kgncq390ic4yzg7qia7rih-libxkbcommon-0.10.0-dev/include -I/build/source/build -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=gnu11 -DWLR_USE_UNSTABLE -Wcast-align -Wdate-time -Werror=format-security -Werror=format=2 -Wendif-labels -Werror=incompatible-pointer-types -Werror=missing-declarations -Werror=overflow -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=implicit-fallthrough=3 -Wformat-nonliteral -Wformat-security -Winit-self -Wmaybe-uninitialized -Wmissing-field-initializers -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wno-missing-field-initializers -Wno-sign-compare -Wno-strict-aliasing -Wno-unused-parameter -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-prototypes -Wtype-limits -Wundef -Wunused-function -fPIC -MD -MQ 'src/25a6634@@phoc@sta/cursor.c.o' -MF 'src/25a6634@@phoc@sta/cursor.c.o.d' -o 'src/25a6634@@phoc@sta/cursor.c.o' -c ../src/cursor.c
In file included from ../src/settings.h:6,
                 from ../src/input.h:8,
                 from ../src/seat.h:5,
                 from ../src/cursor.h:5,
                 from ../src/cursor.c:15:
/nix/store/akcqxqkxivvdx7p17napb0hv7i3hpkv4-libdrm-2.4.100-dev/include/xf86drmMode.h:43:10: fatal error: drm.h: No such file or directory
   43 | #include <drm.h>
      |          ^~~~~~~
compilation terminated.
^M[41/76] Compiling C object 'src/25a6634@@phoc@sta/desktop.c.o'
FAILED: src/25a6634@@phoc@sta/desktop.c.o 
gcc -Isrc/25a6634@@phoc@sta -Isrc -I../src -I/nix/store/vmd834p7b56ba3vmq53rsn5x1yqx77pi-libinput-1.15.5-dev/include -I/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/include -I/nix/store/lh6xk0cgq9pvjdmsq3vh1m829va7ysm3-glib-2.64.1-dev/include/glib-2.0 -I/nix/store/9yfbxys7srn3cgx9jsmfcyakwpkwr9fs-glib-2.64.1/lib/glib-2.0/include -I/nix/store/q6ar7s3ld5l537pkjymwyz92a1cgv1qg-gtk+3-3.24.20-dev/include/gtk-3.0 -I/nix/store/18qmdsyy9f6xxxcxbk8zi7k7gl04z3r7-atk-2.36.0-dev/include/atk-1.0 -I/nix/store/n2amx937mh6k39iadvh567kq4g2ljly7-cairo-1.16.0-dev/include/cairo -I/nix/store/chrni3066nd9li7ccffs9yfqdhlqjdw7-freetype-2.10.1-dev/include/freetype2 -I/nix/store/chrni3066nd9li7ccffs9yfqdhlqjdw7-freetype-2.10.1-dev/include -I/nix/store/pj515z2qsfhwsp3izx0paxfj434hwjqn-gdk-pixbuf-2.40.0-dev/include/gdk-pixbuf-2.0 -I/nix/store/vhc7j53yqsvrr35whr178bh7spxpan6n-gsettings-desktop-schemas-3.36.1/include/gsettings-desktop-schemas -I/nix/store/hnigkkd0xdyl0byb09xxmx55r502acgs-pango-1.44.7-dev/include/pango-1.0 -I/nix/store/kicxnsx8wx5vdjdm1hxq13kkpykkb067-harfbuzz-2.6.4-dev/include/harfbuzz -I/nix/store/jsmq4mkqcklgpwdi34vp3b3g4mfy97z6-gnome-desktop-3.36.2-dev/include/gnome-desktop-3.0 -I/nix/store/5n9qf33f4ynngqjl4fi61ks9ccqhgbgi-pixman-0.38.4/include/pixman-1 -I/nix/store/fh1v53h0y0jcm2rd3cxkd5a8jf9hwp6p-wayland-1.18.0/include -I/nix/store/zvv99h24gav9ydnv1xh9gdlmxdscgkny-wlroots-0.10.1/include -I/nix/store/bwg82bn5m8kgncq390ic4yzg7qia7rih-libxkbcommon-0.10.0-dev/include -I/build/source/build -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=gnu11 -DWLR_USE_UNSTABLE -Wcast-align -Wdate-time -Werror=format-security -Werror=format=2 -Wendif-labels -Werror=incompatible-pointer-types -Werror=missing-declarations -Werror=overflow -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=implicit-fallthrough=3 -Wformat-nonliteral -Wformat-security -Winit-self -Wmaybe-uninitialized -Wmissing-field-initializers -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wno-missing-field-initializers -Wno-sign-compare -Wno-strict-aliasing -Wno-unused-parameter -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-prototypes -Wtype-limits -Wundef -Wunused-function -fPIC -MD -MQ 'src/25a6634@@phoc@sta/desktop.c.o' -MF 'src/25a6634@@phoc@sta/desktop.c.o.d' -o 'src/25a6634@@phoc@sta/desktop.c.o' -c ../src/desktop.c
In file included from ../src/settings.h:6,
                 from ../src/input.h:8,
                 from ../src/seat.h:5,
                 from ../src/desktop.c:36:
/nix/store/akcqxqkxivvdx7p17napb0hv7i3hpkv4-libdrm-2.4.100-dev/include/xf86drmMode.h:43:10: fatal error: drm.h: No such file or directory
   43 | #include <drm.h>
      |          ^~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

Reply:
That looks like a phoc bug.

https://source.puri.sm/Librem5/phoc/-/blob/1ca2c9cfbf09f4b68de378bb97e0a17d3b75ef39/src/settings.h#L6 includes `xf86drmMode.h` from `libdrm` but does not seem to list `libdrm` as a dependency: https://source.puri.sm/Librem5/phoc/-/blob/1ca2c9cfbf09f4b68de378bb97e0a17d3b75ef39/src/meson.build#L28-38

Should be defined [here](https://source.puri.sm/Librem5/phoc/-/blob/1ca2c9cfbf09f4b68de378bb97e0a17d3b75ef39/meson.build#L27-36) as `dependency('libdrm')`.

Reply:
Done: https://source.puri.sm/Librem5/phoc/-/merge_requests/155

I'll update the derivation when it's merged (I hope soon). Otherwise I'll add the patch manually

Reply:
upgraded phoc with the version that include the fix